### PR TITLE
Feature/profile page

### PR DESCRIPTION
**Pull request by flenn77**

### CODE REVIEW

File: package.json
Comment (mouradof) [inline package.json:1]: Why Typescript is imported ?

Comment (flenn77) [reply package.json:1]: it's from develop merge, (maybe to fix it in other branch ?)

File: src/api-user/src/routes/user-routes.js [hunk 1]
```diff
@@ -57,6 +86,12 @@ userRoutes.delete("/user/:id", async c => {
       return c.json({ message: "User not found" }, 404)
     }
 
+    const validPassword = await bcrypt.compare(password, user.passwordHash)
+
+    if (!validPassword) {
+      return c.json({ message: "Password is incorrect" }, 403)
```
Comment (avetisk) [inline src/api-user/src/routes/user-routes.js:92]: Magic number 🌈 

Move your HTTP codes into a file!

File: src/app/components/Navigation.jsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import Link from "next/link"
+
+const Navigation = () => {
+  return (
+    <nav>
+      {/* Insère ici d'autres liens si nécessaire */}
```
Comment (avetisk) [inline src/app/components/Navigation.jsx:6]: Useless comment.

File: src/app/components/Navigation.jsx [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+      <Link href="/profile">Profile</Link>&nbsp;&nbsp;&nbsp;&nbsp;
+      <Link href="/logout">Logout</Link>&nbsp;&nbsp;&nbsp;&nbsp;
+      <Link href="/login">Login</Link>&nbsp;&nbsp;&nbsp;&nbsp;
+      <Link href="/register">Register</Link>&nbsp;&nbsp;&nbsp;&nbsp;
+      <Link href="/home">home</Link>&nbsp;&nbsp;&nbsp;&nbsp;
```
Comment (avetisk) [inline src/app/components/Navigation.jsx:11]: Use CSS!!!!!

File: src/app/components/profile/PasswordChangeForm.jsx [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+      <input
+        id="newPassword"
+        type="password"
+        name="newPassword"
+        value={passwords.newPassword}
+        onChange={handleChange}
+        className="mt-1 block w-full px-3 py-2 border border-gray-300 rounded-md shadow-sm focus:outline-none focus:ring-blue-500 focus:border-blue-500"
+        required
+      />
+    </div>
```
Comment (avetisk) [inline src/app/components/profile/PasswordChangeForm.jsx:32]: Use a form lib like react-hook-form.

File: src/app/components/profile/ProfileHeader.jsx [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+              </li>
+              <li
+                className="px-4 py-2 hover:bg-gray-100 cursor-pointer"
+                onClick={() => router.push("/profile/changePassword")}
+              >
+                Change Password
+              </li>
+              <li className="px-4 py-2 hover:bg-gray-100 cursor-pointer" onClick={handleDeleteAccountRedirect}>
+                Delete Account
+              </li>
```
Comment (avetisk) [inline src/app/components/profile/ProfileHeader.jsx:47]: Use `Link`.

File: src/app/components/profile/delete/DeleteProfileForm.jsx [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+      <div className="flex justify-between w-full">
+        <Button
+          onClick={handleDelete}
+          loading={loading}
+          text="Delete Account"
+          className="bg-red-500 hover:bg-red-600"
+        />
+        <Button
+          // eslint-disable-next-line no-undef
+          onClick={() => router.push(`/profile/${session.id}`)}
```
Comment (avetisk) [inline src/app/components/profile/delete/DeleteProfileForm.jsx:34]: Are you serious???

**Pull request merged**